internal/persist/migration.go

```
@@ -0,0 +1,179 @@
+	sort.Slice(out, func(i, j int) bool { return out[i] < out[j] })
+	return out, nil
+}
+
+func createMatchingBackup(path string, data []byte) error {
+	matched, err := existingBackupMatches(path, data)
+	if err != nil || matched {
+		return err
+	}
+
+	tmp := path + ".tmp"
+	file, err := os.OpenFile(tmp, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0o600)
```

golangci-lint (2.13.2)

[medium] G304: Potential file inclusion via variable

(gosec)